### PR TITLE
Chore/3596 refactor API functions in the frontend

### PR DESCRIPTION
**Pull request by raphj**
#3596

(Note: this was initially intended as a fix for #3518, hence the branch name. I renamed the PR to keep the review history)

This PR also refactors functions used to talk to the API. The fix is in the first commit.

## Checkpoints

**For developers**

- [X] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [X] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [X] Automated tests seem irrelevant

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend_lib/src/action.async.js [hunk 1]
```diff
@@ -1,115 +1,102 @@
 import { FETCH_CONFIG } from './helper.js'
 
-export const putEditContent = (apiUrl, workspaceId, contentId, appSlug, newTitle, newContent, propertiesToAddToBody) => {
-  // INFO - CH - 2019-01-03 - Check the -s added to the app slug. This is and should stay consistent with app features
-  return fetch(`${apiUrl}/workspaces/${workspaceId}/${appSlug}s/${contentId}`, {
+export const baseFetch = (method, url, body, opts) =>
```
Comment (gbonaspetti) [inline frontend_lib/src/action.async.js:3]: `opts` is used somewhere?

File: frontend_app_admin_workspace_user/src/action.async.js [hunk 2]
```diff
@@ -1,114 +1,36 @@
...
 
 export const putUserDisable = (apiUrl, userId) =>
-  fetch(`${apiUrl}/users/${userId}/disabled`, {
-    credentials: 'include',
-    headers: {
-      ...FETCH_CONFIG.headers
-    },
-    method: 'PUT'
-  })
+  baseFetch('GET', `${apiUrl}/users/${userId}/disabled`)
```
Comment (gbonaspetti) [inline frontend_app_admin_workspace_user/src/action.async.js:13]:
```suggestion
  baseFetch('PUT', `${apiUrl}/users/${userId}/disabled`)
```

File: frontend_app_admin_workspace_user/src/action.async.js [hunk 3]
```diff
@@ -1,114 +1,36 @@
...
 
 export const putUserEnable = (apiUrl, userId) =>
-  fetch(`${apiUrl}/users/${userId}/enabled`, {
-    credentials: 'include',
-    headers: {
-      ...FETCH_CONFIG.headers
-    },
-    method: 'PUT'
-  })
+  baseFetch('GET', `${apiUrl}/users/${userId}/enabled`)
```
Comment (gbonaspetti) [inline frontend_app_admin_workspace_user/src/action.async.js:16]:
```suggestion
  baseFetch('PUT', `${apiUrl}/users/${userId}/enabled`)
```

**Comment by grignards** [comment]
I understand your changes but fail to see why it fixes the issue's bug. Could you explain?

File: frontend_app_workspace_advanced/src/action.async.js [hunk 4]
```diff
@@ -1,127 +1,48 @@
...
-    credentials: 'include',
-    headers: { ...FETCH_CONFIG.headers },
-    method: 'PUT',
-    body: JSON.stringify({
-      label: workspace.label,
-      description: newDescription,
-      agenda_enabled: workspace.agenda_enabled
-    })
+  baseFetch('PUT', `${apiUrl}/workspaces/${workspace.workspace_id}`, {
+    description: newDescription
```
Comment (gbonaspetti) [inline frontend_app_workspace_advanced/src/action.async.js:16]: Are you sure that the other parameters weren't necessary?

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

**Comment by raphj** [comment]
@grignards 

So, the fix here: https://github.com/tracim/tracim/pull/3593/files#diff-b7ba43e69b6c83d21cf48be1406d6609R31
Fixes this issue, because the t.map error was caused by two things:

1. the frontend, instead of passing the workspace id given in the url parameter to open a content, gives the current workspace's id
2. it does not handle the 401 error it gets as a result.

This line fixes the first issue. The refactoring that I spent the afternoon on aimed to centralize the error handling to fix the second point, which I haven't had the time to do today. However, I agree with @gbonaspetti (in private messages): it is probably risky to ship Tracim 3.1.1 with this refactoring, so I suggest to merge this PR in develop when all the tests pass, but for tracim 3.1.1 restrict the fix to the one line that makes the frontend use the right workspace id.

File: frontend_app_file/src/container/File.jsx [hunk 5]
```diff
@@ -42,19 +42,19 @@ import {
...
   permissiveNumberEqual
 } from 'tracim_frontend_lib'
 import { PAGE, isVideoMimeTypeAndIsAllowed, DISALLOWED_VIDEO_MIME_TYPE_LIST } from '../helper.js'
 import { debug } from '../debug.js'
 import {
   deleteShareLink,
-  getFileContent,
-  getFileRevision,
   getShareLinksList,
   postShareLinksList,
```
Comment (raphj) [inline frontend_app_file/src/container/File.jsx:57]:
```suggestion
  postShareLinksList
```

File: frontend_app_html-document/src/action.async.js [hunk 6]
```diff
@@ -1,75 +1,26 @@
...
-  fetch(`${apiUrl}/workspaces/${workspaceId}/html-documents/${contentId}`, {
-    credentials: 'include',
-    headers: {
-      ...FETCH_CONFIG.headers
-    },
-    method: 'PUT',
-    body: JSON.stringify({
+  baseFetch('PUT', `${apiUrl}/workspaces/${workspaceId}/html-documents/${contentId}`, {
       label: label,
       raw_content: newContent
```
Comment (raphj) [inline frontend_app_html-document/src/action.async.js:15]:
```suggestion
    label: label,
    raw_content: newContent
```

File: frontend_app_workspace_advanced/src/action.async.js [hunk 7]
```diff
@@ -1,127 +1,48 @@
...
+  baseFetch('GET', `${apiUrl}/workspaces/${workspaceId}/members${showDisabledUser ? '?show_disabled_user=1' : ''}`)
 
-export const getAppList = apiUrl => {
-  return fetch(`${apiUrl}/system/applications`, {
-    credentials: 'include',
-    headers: { ...FETCH_CONFIG.headers },
-    method: 'GET'
-  })
-}
+export const getAppList = () =>
```
Comment (raphj) [inline frontend_app_workspace_advanced/src/action.async.js:6]:
```suggestion
export const getAppList = (apiUrl) =>
```

**Review by gbonaspetti — APPROVED** [review]

**Comment by gbonaspetti** [comment]
Tested on branch.

**Pull request merged**